tools/pipeline-witness/Azure.Sdk.Tools.PipelineWitness/BlobUploadProcessor.cs

```diff
@@ -887,6 +885,56 @@ private async Task UploadTestRunBlobAsync(string account, Build build, TestRun t
             }
         }
 
+        private async Task UploadTestRunResultBlobAsync(string account, Build build, TestRun testRun)
+        {
+            try
+            {
+                var continuationToken = string.Empty;
+                var buildIds = new[] { build.Id };
+                var discoveredTestResults = new List<TestCaseResult>();
+                var blobPath = $"{build.Project.Name}/{testRun.CompletedDate:yyyy/MM/dd}/{testRun.Id}.jsonl";
+                var blobClient = this.testResultContainerClient.GetBlobClient(blobPath);
+
+                if (await blobClient.ExistsAsync())
+                {
+                    this.logger.LogInformation("Skipping existing test results blob for build {BuildId}, test run {RunId}", build.Id, testRun.Id);
+                    return;
+                }
+
+                var data = await testResultsClient.GetTestResultsAsync(build.Project.Id, testRun.Id);
```

Comment:
Just FYI for reviewers, this function does not support a continuation token/paging.